ValueError: Duplicate 'handler', already handled by <DynamicRoute 'handler' [GET] /v1/handler/ -> <function handler at 0x103c85378>

I get a `ValueError` when passing a name to `router.add_route`

```python
        cors.add(
            app.router.add_route(method, full_url, handler, name=name),
            {
                '*': aiohttp_cors.ResourceOptions(
                    allow_credentials=True, expose_headers='*', allow_headers='*'
                )
            }

        )
```

Traceback:

```
    app = create_app(TestConfig)
api/app.py:27: in create_app
    routes.setup(app)
api/routes.py:26: in setup
    allow_credentials=True, expose_headers='*', allow_headers='*'
../../../miniconda/envs/rero/lib/python3.5/site-packages/aiohttp_cors/__init__.py:193: in add
    route, self._preflight_handler)
../../../miniconda/envs/rero/lib/python3.5/site-packages/aiohttp_cors/urldispatcher_router_adapter.py:45: in add_options_method_handler
    self._router.register_route(new_route)
../../../miniconda/envs/rero/lib/python3.5/site-packages/aiohttp/web_urldispatcher.py:459: in register_route
    .format(name, self._routes[name]))
E   ValueError: Duplicate 'should_i_release', already handled by <DynamicRoute 'should_i_release' [GET] /v1/should_i_release/{username}/{repo}/ -> <function should_i_release at 0x103c85378>
```

Removing `name=name` from the call to `app.router.add_route` removes the error.

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -8,6 +8,8 @@ CHANGES
 
 - Rename ``RouterAdapter`` to ``AbstractRouterAdapter``.
 
+- Fix bug with configuring CORS for named routes.
+
 0.1.0 (2015-11-05)
 ------------------
 

--- a/aiohttp_cors/urldispatcher_router_adapter.py
+++ b/aiohttp_cors/urldispatcher_router_adapter.py
@@ -35,14 +35,13 @@ class UrlDistatcherRouterAdapter(AbstractRouterAdapter):
 
     def add_options_method_handler(self, route: web.Route, handler):
         method = "OPTIONS"
-        name = None
 
         if isinstance(route, web.PlainRoute):
             new_route = self._router.add_route(method, route._path, handler)
 
         elif isinstance(route, web.DynamicRoute):
             new_route = web.DynamicRoute(
-                method, handler, name, route._pattern, route._formatter)
+                method, handler, None, route._pattern, route._formatter)
             self._router.register_route(new_route)
 
         elif isinstance(route, web.StaticRoute):
@@ -50,7 +49,7 @@ class UrlDistatcherRouterAdapter(AbstractRouterAdapter):
             # regexp performance is not enough.
             pattern = re.compile("^" + re.escape(route._prefix))
             new_route = web.DynamicRoute(
-                method, handler, name, pattern, "")
+                method, handler, None, pattern, "")
             self._router.register_route(new_route)
 
         else:

--- a/aiohttp_cors/urldispatcher_router_adapter.py
+++ b/aiohttp_cors/urldispatcher_router_adapter.py
@@ -35,13 +35,14 @@ class UrlDistatcherRouterAdapter(AbstractRouterAdapter):
 
     def add_options_method_handler(self, route: web.Route, handler):
         method = "OPTIONS"
+        name = None
 
         if isinstance(route, web.PlainRoute):
             new_route = self._router.add_route(method, route._path, handler)
 
         elif isinstance(route, web.DynamicRoute):
             new_route = web.DynamicRoute(
-                method, handler, route.name, route._pattern, route._formatter)
+                method, handler, name, route._pattern, route._formatter)
             self._router.register_route(new_route)
 
         elif isinstance(route, web.StaticRoute):
@@ -49,7 +50,7 @@ class UrlDistatcherRouterAdapter(AbstractRouterAdapter):
             # regexp performance is not enough.
             pattern = re.compile("^" + re.escape(route._prefix))
             new_route = web.DynamicRoute(
-                method, handler, route.name, pattern, "")
+                method, handler, name, pattern, "")
             self._router.register_route(new_route)
 
         else:

--- a/tests/unit/test_cors_config.py
+++ b/tests/unit/test_cors_config.py
@@ -23,22 +23,23 @@ from aiohttp import web
 from aiohttp_cors import CorsConfig, ResourceOptions
 
 
+def _handler(request):
+    return web.Response(text="Done")
+
+
 class TestCorsConfig(unittest.TestCase):
     """Unit tests for CorsConfig"""
 
     def setUp(self):
-        def handler(request):
-            return web.Response(text="Done")
-
         self.loop = asyncio.new_event_loop()
         self.app = web.Application(loop=self.loop)
         self.cors = CorsConfig(self.app, defaults={
             "*": ResourceOptions()
         })
         self.get_route = self.app.router.add_route(
-            "GET", "/get_path", handler)
+            "GET", "/get_path", _handler)
         self.options_route = self.app.router.add_route(
-            "OPTIONS", "/options_path", handler)
+            "OPTIONS", "/options_path", _handler)
 
     def tearDown(self):
         self.loop.close()

--- a/tests/unit/test_cors_config.py
+++ b/tests/unit/test_cors_config.py
@@ -81,3 +81,31 @@ class TestCorsConfig(unittest.TestCase):
 
         self.assertEqual(len(self.cors._route_config), 1)
         self.assertEqual(len(self.cors._preflight_route_settings), 1)
+
+    def test_plain_named_route(self):
+        """Test adding plain named route."""
+        # Adding CORS routes should not introduce new named routes.
+        self.assertEqual(len(self.app.router.keys()), 0)
+        route = self.app.router.add_route(
+            "GET", "/{name}", _handler, name="dynamic_named_route")
+        self.assertEqual(len(self.app.router.keys()), 1)
+        self.cors.add(route)
+        self.assertEqual(len(self.app.router.keys()), 1)
+
+    def test_dynamic_named_route(self):
+        """Test adding dynamic named route."""
+        self.assertEqual(len(self.app.router.keys()), 0)
+        route = self.app.router.add_route(
+            "GET", "/{name}", _handler, name="dynamic_named_route")
+        self.assertEqual(len(self.app.router.keys()), 1)
+        self.cors.add(route)
+        self.assertEqual(len(self.app.router.keys()), 1)
+
+    def test_static_named_route(self):
+        """Test adding dynamic named route."""
+        self.assertEqual(len(self.app.router.keys()), 0)
+        route = self.app.router.add_static(
+            "/file", "/", name="dynamic_named_route")
+        self.assertEqual(len(self.app.router.keys()), 1)
+        self.cors.add(route)
+        self.assertEqual(len(self.app.router.keys()), 1)

